src/scripts/migrate-traits.js

```diff
@@ -0,0 +1,238 @@
+/**
+ * This script moves the namesAndHandleAppearance and availableForGigs fields from the traits to the main member index
+ * It also does the pre-calculation for the skill-score deduction, to be used in talent search
+ */
+
+require('../../app-bootstrap')
+const _ = require('lodash')
+const helper = require('../common/helper')
+const config = require('config')
+const logger = require('../common/logger')
+
+const constants = require('../../app-constants')
+const fs = require('fs');
+
+const esClient = helper.getESClient()
+
+require('aws-sdk/lib/maintenance_mode_message').suppress = true;
+
+process.on('unhandledRejection', (error) => {
+    console.log(`Unhandled rejection reason: ${JSON.stringify(error, null, 5)}`)
+})
+
+async function getMemberTraits () {
+    console.log('Getting all member traits')
+    const searchResults = {hits:{hits:[]}}
+    const responseQueue = []
+
+    const esQueryTraits = {
+      index: config.ES.MEMBER_TRAIT_ES_INDEX,
+      type: config.ES.MEMBER_TRAIT_ES_TYPE,
+      size: 10000,
+      scroll: '90s'
+    }
+  
+    const response = await esClient.search(esQueryTraits)
+
+    responseQueue.push(response)
+    while (responseQueue.length) {
+        const body = responseQueue.shift()
+        console.log("Loading traits: " + searchResults.hits.hits.length + " of: " + body.hits.total)
+
+        // collect the titles from this response
+        body.hits.hits.forEach(function (hit) {
+            searchResults.hits.hits.push(hit)
+            //searchResults.push(hit._source.quote)
```

Comment:
If this line isn't needed let's remove it.

Reply:
Done